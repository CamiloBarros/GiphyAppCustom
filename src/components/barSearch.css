#input-bar {
    background-color: aliceblue;
    color: black;
    display: block;
    font-size: large;
    height: 30px;
    margin: 0px auto;
    padding: 0px 20px;
    width: 80%;
    border-radius: 20px;
    border: 3px solid transparent;
    text-align: center;
    
}

#input-bar::placeholder {
    font-style: italic;
    font-size: medium;
    
}

#input-bar:hover {
    border: 3px solid tomato;
}

#input-bar:focus {
    border: 3px solid tomato;
    
}