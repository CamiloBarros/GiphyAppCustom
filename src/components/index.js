export * from './bar/BarSearch'
export * from './card/Card'
export * from './grid/Grid'