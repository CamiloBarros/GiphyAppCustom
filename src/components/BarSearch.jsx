import React from 'react'
import './barSearch.css'

export const BarSearch = () => {
    /* Barra de busqueda */
    /* - Se actualizará el estado mientras se va escribiendo (onChange) */
    /* - Se realizará la busqueda al presionar la tecla ENTER */
    /* - Luego de enviar la solicitud se limpiará la barra */

  return (
    <div>
        <input type="text" name="bar-search" id="input-bar" onChange={null} onKeyUp={null} placeholder='Search a item..' />
    </div>
  )
}
