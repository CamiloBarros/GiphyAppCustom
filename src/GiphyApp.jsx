import { BarSearch, Grid } from './components'
import { useFecthApi } from './hooks/useFecthApi'

export const GiphyApp = () => {
  const { gifs, getGif } = useFecthApi()

  return (
    <div>
      <h1>
        <span id='h1-letter'>G</span>iphy<span id='h1-letter'>A</span>pp
      </h1>

      <BarSearch getGif={getGif} />
      <Grid gifs={gifs} />
    </div>
  )
}
