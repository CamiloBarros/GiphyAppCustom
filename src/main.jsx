import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'

import { GiphyApp } from './GiphyApp'
import { Home } from './routes/Home'

import { BrowserRouter, Routes, Route } from 'react-router-dom'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>

    <BrowserRouter>
      <Routes>
        <Route path='/' element={<GiphyApp />} >
          <Route index element={ <Home /> } />
          {/* Routes secondary */}
        </Route>
      </Routes>
    </BrowserRouter>

  </React.StrictMode>
)
