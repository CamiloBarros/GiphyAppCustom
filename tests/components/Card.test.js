import { render, screen } from '@testing-library/react'
import { Card } from '../../src/components'

describe('Pruebas en <Card />', () => {
  const title = 'Saitama'
  const url = 'https://one-punch.com/saitama.jpg'

  test('Debe hacer match con el Snapshot', () => {
    const { container } = render(
      <Card
        source={{
          title,
          url,
        }}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(
      <Card
        source={{
          title,
          url,
        }}
      />
    )

    // expect( screen.getByRole('img').src ).toBe( url )
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('Debe de mostrar el titulo en el componenete', () => {
    render(
      <Card
        source={{
          title,
          url,
        }}
      />
    )

    expect(screen.getByText(title)).toBeTruthy()
  })
})
