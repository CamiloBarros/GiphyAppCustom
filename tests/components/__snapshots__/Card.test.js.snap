// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Card /> Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-container"
    >
      <img
        alt="Saitama"
        class="img-gif"
        src="https://one-punch.com/saitama.jpg"
      />
    </div>
    <div
      class="card-title"
    >
      <span>
        Saitama
      </span>
    </div>
  </div>
</div>
`;
